### PR TITLE
creating cloudformation template for CSPM and Eventbridge

### PR DESCRIPTION
**Pull request by sameer-in**
- creating cloudformation template for CSPM and Eventbridge

### CODE REVIEW

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    Type: String
+    Description: Sysdig ExternalID required for the policy creation
+  SysdigTrustedIdentity:
+    Type: String
+    Description: Sysdig Trusted identity required for policy creation
+  SysdigEBName:
+    Type: String
+    Default: "sysdig"
+    Description: Unique name for role and policy
+  SysdigExternalID:
```
Comment (nkraemer-sysdig) [inline templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml:18]: As caught by linting, this is a duplicate param

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    Type: String
+    Default: "SysdigAgentlessRole"
+    Description: Unique role for monitoring AWS accounts
+  SysdigExternalID:
+    Type: String
+    Description: Sysdig ExternalID required for the policy creation
+  SysdigTrustedIdentity:
+    Type: String
+    Description: Sysdig Trusted identity required for policy creation
+  SysdigEBName:
```
Comment (nkraemer-sysdig) [inline templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml:14]: I'm not sure if this role should be configurable. If we make it configurable, we will need an additional input to collect this role name in the sysdig UI.

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: IAM Role for Agentless
+Parameters:
+  SysdigRoleName:
+    Type: String
+    Default: "SysdigAgentlessRole"
```
Comment (nkraemer-sysdig) [inline templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml:6]: Can we check that this is the right default name with product? Does it match the default provisioned by the TF modules?

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml [hunk 4]
```diff
@@ -0,0 +1,87 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: IAM Role for Agentless
```
Comment (nkraemer-sysdig) [inline templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml:2]: This template does more than just IAM role

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    Type: String
+    Default: "sysdig"
+    Description: Unique name for role and policy
+  SysdigExternalID:
+    Type: String
+    Description: Sysdig ExternalID required for the policy creation
+  SysdigEventBusARN:
+    Type: String
+    Description: Sysdig event bus ARN
+  DeadLetterQueueARN:
```
Comment (nkraemer-sysdig) [inline templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml:24]: @cgeers do we have a plan with the FC ingestion team about DLQs?

File: templates_cspm/CloudAgentlessRole.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+Resources:
+  CloudAgentlessRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      RoleName: !Ref SysdigRoleName
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          -
+            Effect: "Allow"
```
Comment (nkraemer-sysdig) [inline templates_cspm/CloudAgentlessRole.yaml:24]: Nit: Do we need this newline?

File: templates_eventbridge/EventBridge.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: EventBridgeRole and EventBridgeRole for falco cloud
```
Comment (nkraemer-sysdig) [inline templates_eventbridge/EventBridge.yaml:2]: Typo here

File: templates_eventbridge/EventBridge.yaml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+Parameters:
+  SysdigName:
+    Type: String
+    Default: "sysdig"
```
Comment (nkraemer-sysdig) [inline templates_eventbridge/EventBridge.yaml:6]: Again, should this role name be consistent?

File: templates_cspm/CloudAgentlessRole.yaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: IAM Role for Agentless
+Parameters:
+  SysdigRoleName:
+    Type: String
+    Default: "SysdigAgentlessRole"
```
Comment (nkraemer-sysdig) [inline templates_cspm/CloudAgentlessRole.yaml:6]: Is this default consistent?

File: templates_eventbridge/EventBridge.yaml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+        - Id: !Ref SysdigName
+          Arn: !Ref SysdigEventBusARN
+          RoleArn: !GetAtt
+            - EventBridgeRole
+            - Arn
+          DeadLetterConfig:
+            !If
+              - IsDeadLetterQueueARNNotEmpty
+              -
+                Arn: !Ref DeadLetterQueueARN
```
Comment (nkraemer-sysdig) [inline templates_eventbridge/EventBridge.yaml:59]: Do we need this newline?

**Review by nkraemer-sysdig — COMMENTED** [review]
One additional ask: Since these templates directly control what is displayed in the AWS console (Defaults, param names, descriptions etc) can you please:
1) Attach a screenshot of each template as it is shown in the AWS consone
2) Review these with Product/design to confirm the copy is correct

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml (hunk 1 above)
Comment (sameer-in) [reply templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml]: removed the duplicate

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml (hunk 4 above)
Comment (sameer-in) [reply templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml]: added correct description.

File: templates_eventbridge/EventBridge.yaml (hunk 7 above)
Comment (sameer-in) [reply templates_eventbridge/EventBridge.yaml]: Fixed the typo here.

File: templates_cspm/CloudAgentlessRole.yaml (hunk 6 above)
Comment (sameer-in) [reply templates_cspm/CloudAgentlessRole.yaml]: Removed this newline. Its not needed

File: templates_cspm/CloudAgentlessRole.yaml (hunk 9 above)
Comment (sameer-in) [reply templates_cspm/CloudAgentlessRole.yaml]: renamed it to default created by terraform.

File: templates_eventbridge/EventBridge.yaml (hunk 10 above)
Comment (sameer-in) [reply templates_eventbridge/EventBridge.yaml]: Nope. removed this as its not necessary

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml (hunk 2 above)
Comment (sameer-in) [reply templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml]: This is the `name` variable in terraform for EB. Its optional and  `default` value is set to `sysdig`. But customers can override this value

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml (hunk 3 above)
Comment (sameer-in) [reply templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml]: checked the terraform and set the default value to `sfc-cspm-role`

File: templates_eventbridge/EventBridge.yaml (hunk 8 above)
Comment (sameer-in) [reply templates_eventbridge/EventBridge.yaml:6]: Its renamed to SysdigEBName

File: templates_cspm/CloudAgentlessRole.yaml (hunk 9 above)
Comment (sameer-in) [reply templates_cspm/CloudAgentlessRole.yaml]: renamed it.

**Comment by sameer-in** [comment]
Attaching screenshots for every template
CSPM
EB
cspm-EB

**Comment by sameer-in** [comment]
These are the latest screenshots
cspm
EB
cspm-eb

File: templates_eventbridge/EventBridge.yaml [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    Description: Sysdig ExternalID required for the policy creation
+  SysdigTrustedIdentity:
+    Type: String
+    Description: The Role in Sysdig’s AWS Account with permissions to your account    
+  SysdigEventBusARN:
+    Type: String
+    Description: The destination in Sysdig’s AWS account where your events are sent
+  DeadLetterQueueARN:
+    Type: String
+    Description: DeadLetterQueueARN ARN
```
Comment (nkraemer-sysdig) [inline templates_eventbridge/EventBridge.yaml:47]: This parameter can be removed, we aren't supporting DLQ yet

**Review by nkraemer-sysdig — COMMENTED** [review]
We can probably drop the `Sysdig` prefix from all the parameters, since it will just make the URL more verbose when it includes all the parameters

File: templates_eventbridge/EventBridge.yaml (hunk 11 above)
Comment (sameer-in) [reply templates_eventbridge/EventBridge.yaml]: done. removed it.

File: templates_cspm_eventbridge/FullInstall.yaml [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    Description: Sysdig ExternalID required for the policy creation
+  SysdigTrustedIdentity:
+    Type: String
+    Description: The Role in Sysdig’s AWS Account with permissions to your account    
+  SysdigEventBusARN:
+    Type: String
+    Description: The destination in Sysdig’s AWS account where your events are sent
+  DeadLetterQueueARN:
+    Type: String
+    Description: DeadLetterQueueARN ARN
```
[the remark this answers is not in the document]
Comment (sameer-in) [reply templates_cspm_eventbridge/FullInstall.yaml]: done. removed it.

**Comment by sameer-in** [comment]
@nkraemer-sysdig removed the `Sysdig` prefix from parameter names.

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml (hunk 3 above)
Comment (sameer-in) [reply templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml]: renamed it to `sysdig-secure` here and in terraform

File: templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml (hunk 5 above)
Comment (sameer-in) [reply templates_apprunner/templates_cspm_eventbridge/FullInstall.yaml]: @cgeers and @nkraemer-sysdig we are removing DLQs from CFT and Terraform.

File: templates_cspm_eventbridge/FullInstall.yaml [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+Parameters:
+  RoleName:
+    Type: String
+    Default: "sysdig-secure"
+    Description: The read-only IAM Role that Sysdig will create
+  EventBridgeRoleName:
+    Type: String
+    Default: "sysdig"
+    Description: Unique name for role and policy
```
Comment (nkraemer-sysdig) [inline templates_cspm_eventbridge/FullInstall.yaml:39]: I'm not sure this needs to be configurable or if it should be hardcoded. @cgeers the EB policy/rules don't key off the EB Rule/Role name right?

Comment (nkraemer-sysdig) [reply templates_cspm_eventbridge/FullInstall.yaml:39]: Can we use `sysdig-threat-detection` in the resource creation and remove this as a parameter?

Comment (sameer-in) [reply templates_cspm_eventbridge/FullInstall.yaml:39]: default values is provided and removed it from parameters list. Pls check the screenshot for full-install @nkraemer-sysdig

File: templates_cspm/OrgCloudAgentlessRole.yaml [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+      StackSetName: AgentLessRoleStackSet
+      Description: Create a Role in all regions
+      PermissionModel: SELF_MANAGED
+      StackInstancesGroup:
+        - DeploymentTargets:
+            Accounts:
+              - !Ref "AWS::AccountId"
+          Regions:
+            - eu-west-2
+            - eu-west-1
```
Comment (nkraemer-sysdig) [inline templates_cspm/OrgCloudAgentlessRole.yaml:45]: Why only these two regions?

File: templates_cspm/OrgCloudAgentlessRole.yaml [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+  RoleStackSet:
+    Type: AWS::CloudFormation::StackSet
+    Properties:
+      StackSetName: AgentLessRoleStackSet
+      Description: Create a Role in all regions
+      PermissionModel: SELF_MANAGED
+      StackInstancesGroup:
+        - DeploymentTargets:
+            Accounts:
+              - !Ref "AWS::AccountId"
```
Comment (nkraemer-sysdig) [inline templates_cspm/OrgCloudAgentlessRole.yaml:42]: Will this deploy to all accounts in the org?

File: templates_eventbridge/OrgEventBridge.yaml [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+  SysdigEventBusARN:
+    Type: String
+    Description: Sysdig event bus ARN
+  DeadLetterQueueARN:
+    Type: String
+    Description: DeadLetterQueueARN ARN
+
+Conditions:
+  IsDeadLetterQueueARNNotEmpty:
+    !Not [!Equals [ !Ref DeadLetterQueueARN, ""]]
```
Comment (nkraemer-sysdig) [inline templates_eventbridge/OrgEventBridge.yaml:20]: This needs to be removed

File: templates_eventbridge/OrgEventBridge.yaml [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+    Properties:
+      Description: Create a Role in all regions
+      PermissionModel: SELF_MANAGED
+      StackInstancesGroup:
+        - DeploymentTargets:
+            Accounts:
+              - !Ref "AWS::AccountId"
+          Regions:
+            - eu-west-2
+            - eu-west-1
```
Comment (nkraemer-sysdig) [inline templates_eventbridge/OrgEventBridge.yaml:72]: Same here

File: templates_eventbridge/OrgEventBridge.yaml (hunk 16 above)
Comment (sameer-in) [reply templates_eventbridge/OrgEventBridge.yaml]: removing the org templates for now.

File: templates_eventbridge/OrgEventBridge.yaml (hunk 17 above)
Comment (sameer-in) [reply templates_eventbridge/OrgEventBridge.yaml]: removing the org templates for now.

File: templates_cspm/OrgCloudAgentlessRole.yaml (hunk 15 above)
Comment (sameer-in) [reply templates_cspm/OrgCloudAgentlessRole.yaml]: removing the org templates for now.

File: templates_cspm/OrgCloudAgentlessRole.yaml (hunk 14 above)
Comment (sameer-in) [reply templates_cspm/OrgCloudAgentlessRole.yaml]: removing the org templates for now.

**Review by hayk99 — APPROVED** [review]
checked the GH workflows and the makefiles and they seem to be correct
✅

**Review by nkraemer-sysdig — APPROVED** [review]

**Comment by nkraemer-sysdig** [comment]
Looks good Sameer, thanks for going through so many rounds of iteration

**Pull request merged**